### ISSUE TITLE
[Feature]: Whatsapp flows to handle conversation flow

### ISSUE THREAD

**Issue by misingo255** [Tanzania-AI-Community/twiga#62]
**Is your feature request related to a problem? Please describe.**
The chat flow seems to be too long and tiresome.

**Describe the solution you'd like**
I suggest using WhatsApp flows to handle conversation flow and selection of attributes (e.g. Topic, Class, Type Of Question, number of questions to be generated etc.) which gives the user a form like experience which is attractive and fun.

**Describe alternatives you've considered**  
One alternative could be using quick replies or buttons in WhatsApp to guide the user through the process. For instance, instead of asking open-ended questions in the chat flow, the chatbot can present pre-defined options (e.g., buttons for topics, question types, or classes). This approach simplifies interaction but may not feel as interactive or visually appealing as the form-like experience provided by WhatsApp flows.

**Additional context**  
Including multimedia elements like images, GIFs, or videos in the WhatsApp flow could make the experience more engaging and less tiresome. For example, a short video explaining how to use the chatbot or visual representations of different question types could enhance user understanding.

It might also help to analyze user feedback or engagement metrics to determine where the chat flow becomes tiresome. Adjustments could then focus on those specific pain points.

NB:

Check out the following about WhatsApp flows:-

- https://www.youtube.com/watch?v=GG8JfmVkj08
- https://www.youtube.com/watch?v=TOluedENQgQ

CC:

- @jurmy24 
- @fredygerman

**Comment by jurmy24** [comment]
Hey @misingo255! Awesome suggestion. We have actually already replaced the previous way of onboarding new users with WhatsApp flows (we have a pending Pull Request to integrate it). So you're exactly right that it's a much better solution than what you see in the demo at `docs/twiga.gif`, which is an older version. 

Regarding using flows for the actual interactions (i.e. teachers requesting some exercises for example). Our field studies suggest that typical public school teachers in Tanzania are mostly comfortable using WhatsApp for texting their colleagues and friends/family. We want to make the experience of using it very similar to that. So we're working on making the chatbot as intelligent as possible so that the teachers hardly need to know what "features" it has and instead it provides them with examples. But perhaps when a new teacher joins we need a better process for making it clear to them what they can use Twiga for.

**Comment by fredygerman** [comment]
I agree with @jurmy24 and for now to help the new teacher understand the chatbot better we can even send a tutorial or demo video and / or document after they complete the onboarding.

**Comment by jurmy24** [comment]
Though I quite like the idea of sending a flow if the teacher requests an exercise or exam so that they can fill in specifically what they want and what textbooks/topics, etc... they want it to make use of. 

For example, it would be nice if they could just tick a box saying they want the exam as a PDF and then we send that to them on WhatsApp.

**Comment by misingo255** [comment]
@jurmy24 This is cool. I can't wait to see the new update. I'm looking forward to it.

Also, be free to check me if you need any assistance.

Thanks.

**Comment by jurmy24** [comment]
For sure! Thanks for the help. I'll tag you if I think there's any task you might want to work on. Let me know what your interests are and I'm sure we can find something.

**Comment by jurmy24** [comment]
Note to self: contact Meta for support
